### PR TITLE
Rename type provider to f0 (function/0)

### PR DESCRIPTION
**Pull request by sjanota**

### CODE REVIEW

File: src/function_utils.erl [hunk 1]
```diff
@@ -32,11 +32,11 @@
 -type callback(T) ::
     {module(), atom(), Args :: list()} |
     {fun(), Args :: list()} |
-    provider(T).
+    f0(T).
 
 -export_type([
-    provider/1,
-    provider/0,
+    f0/1,
```
Comment (kubaodias) [inline src/function_utils.erl]: Should this type be exported?

Comment (sjanota) [reply src/function_utils.erl]: Actually this is pretty useful shortcut for `fun(() -> any())`. It probably is not used anywhere right now, but definitely could be. Helps with readability imho. I thought about adding also f1 as it is another popular type of function.

Comment (sjanota) [reply src/function_utils.erl]: Your call, I have no strong opinion about it

Comment (kubaodias) [reply src/function_utils.erl]: Sounds reasonable

**Pull request merged**